secboot/secboot_tpm.go

```diff
@@ -309,7 +321,14 @@ func SealKeys(keys []SealKeyRequest, params *SealKeysParams) error {
 
 	if params.TPMProvision {
 		// Provision the TPM as late as possible
-		if err := tpmProvision(tpm, params.TPMLockoutAuthFile); err != nil {
+		existingLockoutAuth := params.TPMPartialReprovision
+		if err := tpmProvision(tpm, params.TPMLockoutAuthFile, existingLockoutAuth); err != nil {
+			return err
+		}
+	}
+	if params.TPMPartialReprovision {
```

Comment:
from the description above it sounded like this should be inside the if for TPMProvision? fo we need to check that TPMProvision is set?